### ISSUE TITLE
[Core feature request] `run` multiline string command

### ISSUE THREAD

**Issue by benkeil** [typesafegithub/github-workflows-kt#753]
**What feature do you need?**

If the `run` command, contains `\n` it should be also a multiline string in the resulting yaml for better readability.

**Do you have an example usage?**

```kotlin
run(
    name = "upload package to s3",
    workingDirectory = workingDir,
    command = """
        VERSION=$(cat package.json | jq -r '.version')
        BUCKET_NAME=$(aws ssm get-parameter --name $bucketNameSsmParameter | jq -r '.Parameter.Value')
        aws s3 cp dist s3://$bucketName/apps/accountinfo/$version/ --recursive 
        echo $version > VERSION
        aws s3 cp VERSION s3://$bucketName/apps/accountinfo/
    """.trimIndent(),
)
```

**Is there a workaround for not having this feature? If yes, please describe it.**

no impact, just not as readable as multiline

**Comment by krzema12** [comment]
Yes, I agree that ideally the new lines should be present in the YAML.

Here's a related issue: https://github.com/typesafegithub/github-workflows-kt/issues/616 where we set max line length for snakeyaml (our YAML emitting library) to max, but it was rather a mitigation/workaround than a proper solution:
https://github.com/typesafegithub/github-workflows-kt/blob/276c94f5b2375ac68222fb40af0e145d9eca4bc0/library/src/main/kotlin/it/krzeminski/githubactions/yaml/ObjectToYaml.kt#L23-L25

It was likely causes by a bug in snakeyaml. We track reporting this and other issues to the maintainer in https://github.com/typesafegithub/github-workflows-kt/issues/634. Once this is fixed, we can consider going back to having proper new lines in the emitted YAML.

**Comment by krzema12** [comment]
I'm trying to reproduce now and I can't - see e.g. this multiline string present in the YAML:
https://github.com/typesafegithub/github-workflows-kt/blob/c2d5065f6e64422d24d7452cbb82f4e9048ce88e/.github/workflows/bindings-server.yaml#L70-L72

Please reopen if you find a repro.
